backend/src/apiserver/main.go

```diff
@@ -235,10 +267,42 @@ func startHttpProxy(resourceManager *resource.ResourceManager) {
 	// Register a handler for Prometheus to poll.
 	topMux.Handle("/metrics", promhttp.Handler())
 
+	go startWebhookHTTPSProxy(controllerClient)
+
 	http.ListenAndServe(*httpPortFlag, topMux)
 	glog.Info("Http Proxy started")
 }
 
+func startWebhookHTTPSProxy(controllerClient ctrlclient.Client) {
+	glog.Info("Starting Webhook HTTPS Proxy")
+
+	tlsCertPath := os.Getenv("WEBHOOK_TLS_CERT")
+	tlsKeyPath := os.Getenv("WEBHOOK_TLS_KEY")
+
+	topMux := mux.NewRouter()
+
+	pvValidateWebhook, err := webhook.NewPipelineVersionWebhook(controllerClient)
+	if err != nil {
+		log.Fatalf("Failed to instantiate the Kubernetes webhook: %v", err)
+	}
+
+	topMux.Handle("/webhooks/validate-pipelineversion", pvValidateWebhook)
+
+	if common.FileExists(tlsCertPath) && common.FileExists(tlsKeyPath) {
```

Comment:
This line assumes that `tlsCertPath` and `tlsKeyPath` are always set which it's possible they aren't.